tools/build_script_generator/scons/resources/SConscript.in

```diff
@@ -122,13 +122,13 @@ env.AppendUnique(CPPPATH=[
 	"/usr/local/include",
 %% endif
 %% for path in include_paths | sort
-    abspath("{{ path | relocate }}"),
+    abspath("{{ path | relocate | modm.windowsify }}"),
```

Comment:
```suggestion
    abspath("{{ path | relocate | modm.windowsify(escape_level=2) }}"),
```
I guess it's ok?

Reply:
I guess it is "ok", not sure it buys us anything though, it's basically more hacks :D

Reply:
Yeah, but this isn't just a Python problem anymore now with gdbinit.
I think the filter would also have to be more robust:
1. Collapse all duplicate `\` and `/`
2. Convert all `/` to one `\`.
3. Convert all `\` to `\` * escape_level.

Reply:
How about calling [`normpath`](https://docs.python.org/3/library/os.path.html#os.path.normpath) then expanding the `\`s?

Also, shouldn't `escape_level` be the number of doublings? So that
```
escape_level=0 => "\" (leave as is)
escape_level=1 => "\\" (escape once)
escape_level=2 => "\\\\" (escape twice)
```
I don't think for example turning a `\` into `\\\` would ever make sense.

Reply:
Yes, that's much better!